### PR TITLE
Refactor MTE-3854 Menu Refactor for XCUITests (WIP)

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3854)

## :bulb: Description
Work in progress: Update `BrowserTabMenu` for the upcoming menu refactor work. 💪🏼 

Todo:
* Transplant *all* actions and transitions from the old `BrowserTabMenu` to the new menu.

Current status and known issues: https://docs.google.com/document/d/17LxnKAX_g-ZiDMZSO_zcRbA17zMtUIH7L6XS79wjfh0/edit?tab=t.0

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by isabelrios** [comment]
@clarmso there is a unit test failing that blocks this PR to be run in Bitrise to see the Smokestest results:
```
testTelemetryForSiteMenu()
ClientTests / TabToolbarHelperTests
```
We would need it to be fixed or disabled so that we can see the progress on the UI tests.

File: firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan [hunk 1]
```diff
@@ -119,6 +119,7 @@
         "ShortcutRouteTests\/testOpenLastBookmarkShortcutWithInvalidUrl()",
         "TabManagerTests\/testDeleteSelectedTab()",
         "TabManagerTests\/testPrivatePreference_togglePBMDeletesPrivate()",
+        "TabToolbarHelperTests\/testTelemetryForSiteMenu()",
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan:122]: This unit test is failing consistently on this branch.

**Comment by clarmso** [comment]
Not sure why Swiftlint fail. After I cherry pick the changes from `main` today, *all* smoke tests pass: https://github.com/mozilla-mobile/firefox-ios/pull/23239

**Comment by clarmso** [comment]
Newer PR supersedes this branch.

**Pull request closed without merging**